oauth2/oauth2-jose/src/test/java/org/springframework/security/oauth2/jwt/NimbusReactiveJwtDecoderTests.java

```diff
@@ -205,4 +219,70 @@ public void setClaimSetConverterWhenNullThrowsIllegalArgumentException() {
 		assertThatCode(() -> this.decoder.setClaimSetConverter(null))
 				.isInstanceOf(IllegalArgumentException.class);
 	}
+
+	@Test
+	public void constructorWhenSecretKeyNullThenThrowsIllegalArgumentException() {
+		assertThatThrownBy(() -> new NimbusReactiveJwtDecoder(null, JwsAlgorithms.HS256))
+				.isInstanceOf(IllegalArgumentException.class)
+				.hasMessage("secretKey cannot be null");
+	}
+
+	@Test
+	public void constructorWhenJwsAlgorithmNullThenThrowsIllegalArgumentException() {
+		assertThatThrownBy(() -> new NimbusReactiveJwtDecoder(secretKey(), null))
+				.isInstanceOf(IllegalArgumentException.class)
+				.hasMessage("jwsAlgorithm cannot be empty");
+	}
+
+	@Test
+	public void constructorWhenJwsAlgorithmInvalidThenThrowsIllegalArgumentException() {
+		assertThatThrownBy(() -> new NimbusReactiveJwtDecoder(secretKey(), "invalid"))
+				.isInstanceOf(IllegalArgumentException.class)
+				.hasMessage("Invalid jwsAlgorithm 'invalid'. It must be HS256, HS384 or HS512.");
+	}
+
+	@Test
+	public void decodeWhenSecretKeyThenSuccess() throws Exception {
+		SecretKey secretKey = secretKey();
+		String jwsAlgorithm = JwsAlgorithms.HS256;
+		JWTClaimsSet claimsSet = new JWTClaimsSet.Builder()
+				.subject("test-subject")
+				.expirationTime(Date.from(Instant.now().plusSeconds(60)))
+				.build();
+		SignedJWT signedJWT = signedJwt(secretKey, jwsAlgorithm, claimsSet);
+
+		this.decoder = new NimbusReactiveJwtDecoder(secretKey, jwsAlgorithm);
+
+		Jwt jwt = this.decoder.decode(signedJWT.serialize()).block();
+		assertThat(jwt.getSubject()).isEqualTo("test-subject");
+	}
+
+	@Test
+	public void decodeWhenSecretKeyAndAlgorithmMismatchThenThrowsJwtException() throws Exception {
+		SecretKey secretKey = secretKey();
+		String jwsAlgorithm = JwsAlgorithms.HS256;
+		JWTClaimsSet claimsSet = new JWTClaimsSet.Builder()
+				.subject("test-subject")
+				.expirationTime(Date.from(Instant.now().plusSeconds(60)))
+				.build();
+		SignedJWT signedJWT = signedJwt(secretKey, jwsAlgorithm, claimsSet);
+
+		this.decoder = new NimbusReactiveJwtDecoder(secretKey, JwsAlgorithms.HS512);
+
+		assertThatThrownBy(() -> this.decoder.decode(signedJWT.serialize()).block())
+				.isInstanceOf(JwtException.class);
+	}
+
+	private SecretKey secretKey() throws NoSuchAlgorithmException {
+		KeyGenerator keyGenerator = KeyGenerator.getInstance("AES");
+		keyGenerator.init(256);
+		return keyGenerator.generateKey();
```

Comment:
It'd be nice to just use a static key here and use `SecretKeySpec`. It would make the test faster.